### DISCUSSION TITLE
How to manually cancel a helm upgrade while in progress?

### DISCUSSION THREAD

**Discussion by cep21** [fluxcd/flux2#2816] [Q&A] [answered]
We had a helm chart with a 45 minute timeout.  It's currently installing and we know why it's stuck `Deployment is not ready: X/Y. 0 out of 1 expected pods are ready` and will never resolve until the install ends and we change some values.  I would love to run a `flux` command to stop this helm install early.  `flux suspend` doesn't preemptively stop the current helm install.  Is there a way to trigger `flux` helm operator to know the currently attempted install of a chart won't resolve and timeout now.

**Comment by stefanprodan** [comment, marked as answer]
There is no way of stopping it without restarting helm-controller with `kubectl -n flux-system delete po helm-controller-xxxx`.

Please create an issue for this feature request in the https://github.com/fluxcd/helm-controller repository.

**Reply by cep21** [reply]
Thanks for the answer.  We've noticed that if the helm controller restarts while it's trying to install or upgrade a chart, the helm chart gets stuck in the upgrade phase and never completes when the helm controller starts back up, so deleting the pod would require manually fixing the broken install, right?

**Reply by stefanprodan** [reply]
Yes, we're going to fix this deadlock in the next helm-controller minor release.
